yt/python/yt/wrapper/spec_builders.py

```diff
@@ -1702,13 +1702,21 @@ def prepare(self, client=None):
 
     def _do_build_mapper(self, spec, client):
         if isinstance(spec["mapper"], UserJobSpecBuilder):
-            command = spec["mapper"]._spec["command"]
+            mapper_spec = spec["mapper"]._spec
         else:
-            command = spec["mapper"].get("command")
+            mapper_spec = spec["mapper"]
+
+        command = mapper_spec.get("command")
+        output_streams = mapper_spec.get("output_streams")
 
         intermediate_stream_count = 1
         if isinstance(command, TypedJob) and command.get_intermediate_stream_count() is not None:
             intermediate_stream_count = command.get_intermediate_stream_count()
+            if output_streams is not None:
```

Comment:
Oh, I've just realized that there is `command.get_intermediate_stream_count() is not None` here.

So this check should be outside of this section.
```python
if isinstance(command, TypedJob and output_streams is not None:
    raise YtError("Output streams cannot be specified explicitly for a typed map job, use get_intermediate_stream_count and prepare_operation methods instead")
```

@dim-an I can fix it myself if needed. Or is it more convenient for you to fix this yourself in an internal review?

Reply:
Maybe something like this will be better:

```python
if isinstance(command, TypedJob):
    intermediate_stream_count = command.get_intermediate_stream_count() or 1
    if output_streams is not None:
        # ...
elif <...>:
    # ...
```